File: python/src/core/rectangle.py
from core.qgon import QGon
from core.point2d import Point2D

class Rectangle(QGon):
	def __init__(self, p):
		self.n = 4
		self.p = p

	def __str__(self):
		st = f"Rectangle: "
		for i in range(len(self.p)):
			st += f"{str(self.p[i])}, "
		st = st[:len(st)-2]
		return st

	def square(self):
		return Point2D.sub(self.p[1], self.p[0]).abs() * Point2D.sub(self.p[2], self.p[1]).abs()

File: csharp/WorkOOP/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WorkOOP
{
    // Реализация класса Rectangle – прямоугольник.
    internal class Rectangle: QGon
    {
        /*
         * Rectangle – конструктор, описывающий инициализацию объекта. 
         * 
         * На вход принимает массив точек в порядке обхода по прямоугольнику.
         */
        public Rectangle(Point2D[] p) : base(p) { }

        /*
         * Переопределение под прямоугольник функции square()
         */
        new public double square() => Point2D.sub(p[1], p[0]).abs() * Point2D.sub(p[2], p[1]).abs();
    }
}
